unity-renderer/Assets/Rendering/ProceduralSkybox/ToolProceduralSkybox/Scripts/SkyboxUtils.cs

```diff
@@ -43,60 +46,69 @@ public static class SkyboxShaderUtils
+                shaderLayersProperties[i].Add("_timeFrame_" , Shader.PropertyToID("_timeFrame_" + i));
+                shaderLayersProperties[i].Add("_rowAndCollumns_" , Shader.PropertyToID("_rowAndCollumns_" + i));
 
-                shaderLayersProperties.Add("_lightIntensity_" + i, Shader.PropertyToID("_lightIntensity_" + i));
-                shaderLayersProperties.Add("_normalIntensity_" + i, Shader.PropertyToID("_normalIntensity_" + i));
+                shaderLayersProperties[i].Add("_lightIntensity_" , Shader.PropertyToID("_lightIntensity_" + i));
+                shaderLayersProperties[i].Add("_normalIntensity_" , Shader.PropertyToID("_normalIntensity_" + i));
 
-                shaderLayersProperties.Add("_distortIntAndSize_" + i, Shader.PropertyToID("_distortIntAndSize_" + i));
-                shaderLayersProperties.Add("_distortSpeedAndSharp_" + i, Shader.PropertyToID("_distortSpeedAndSharp_" + i));
+                shaderLayersProperties[i].Add("_distortIntAndSize_" , Shader.PropertyToID("_distortIntAndSize_" + i));
+                shaderLayersProperties[i].Add("_distortSpeedAndSharp_" , Shader.PropertyToID("_distortSpeedAndSharp_" + i));
 
 
-                shaderLayersProperties.Add("_particlesMainParameters_" + i, Shader.PropertyToID("_particlesMainParameters_" + i));
-                shaderLayersProperties.Add("_particlesSecondaryParameters_" + i, Shader.PropertyToID("_particlesSecondaryParameters_" + i));
+                shaderLayersProperties[i].Add("_particlesMainParameters_" , Shader.PropertyToID("_particlesMainParameters_" + i));
+                shaderLayersProperties[i].Add("_particlesSecondaryParameters_" , Shader.PropertyToID("_particlesSecondaryParameters_" + i));
 
-                shaderLayersProperties.Add("_tilingAndOffset_" + i, Shader.PropertyToID("_tilingAndOffset_" + i));
-                shaderLayersProperties.Add("_speedAndRotation_" + i, Shader.PropertyToID("_speedAndRotation_" + i));
+                shaderLayersProperties[i].Add("_tilingAndOffset_" , Shader.PropertyToID("_tilingAndOffset_" + i ));
+                shaderLayersProperties[i].Add("_speedAndRotation_" , Shader.PropertyToID("_speedAndRotation_" + i));
             }
 
         }
 
-        public static int GetLayerProperty(string name)
+        public static int GetLayerProperty(string name, int layer)
         {
+            if (layer >= TOTAL_SKYBOX_LAYERS)
+            {
+                throw new ArgumentException($"Maximum Skybox Layers is {TOTAL_SKYBOX_LAYERS}");
+            }
+
             int propertyInt;
-            if (shaderLayersProperties == null || shaderLayersProperties.Count <= 0)
+            if (shaderLayersProperties is not { Length: > 0 })
             {
                 CacheShaderProperties();
             }
 
-            if (!shaderLayersProperties.ContainsKey(name))
+            if (!shaderLayersProperties[layer].TryGetValue(name, out int val))
```

Comment:
I would revert this `if` just for the sake of readability

Reply:
@popuz what do you mean? I changed it to a TryGetValue so that we obtain the value without having to index inside the dictionary twice.

Reply:
in most cases, it is more clear to us to read from positive. So I suggested
```
if(shaderLayersProperties[layer].TryGetValue(name, out int val))
 ...
else
 ...
```

Reply:
Agreed